python/ray/data/datasource/parquet_datasource.py

```diff
@@ -246,6 +251,36 @@ def get_read_tasks(self, parallelism: int) -> List[ReadTask]:
 
         return read_tasks
 
+    def _estimate_decompression_multiplier(self) -> float:
+        """Return an estimate of the decompression multipler.
+
+        To avoid OOMs, it is safer to return an over-estimate than an underestimate.
+        """
+        # Sample a few rows from the row group to estimate the ratio.
+        # TODO(ekl/cheng) take into account column pruning.
+        num_samples = min(
+            PARQUET_COMPRESSION_RATIO_ESTIMATE_NUM_SAMPLES, len(self._pq_ds.pieces)
```

Comment:
if number of files is less than `PARQUET_COMPRESSION_RATIO_ESTIMATE_NUM_SAMPLES`, do we still want to do sampling? I am worried about it introduces unnecessary extra overhead when reading just a few files.

Reply:
I think we'd want an accurate estimate in each case. The overhead should be fixed and reasonably small in the number of files is low.